src/shared/netif-naming-scheme.c

```diff
@@ -101,3 +103,78 @@ static const char* const alternative_names_policy_table[_NAMEPOLICY_MAX] = {
 };
 
 DEFINE_STRING_TABLE_LOOKUP(alternative_names_policy, NamePolicy);
+
+static int naming_sysattr_allowed_by_default(sd_device *dev) {
+        int r;
+
+        assert(dev);
+
+        r = device_get_property_bool(dev, "ID_NET_NAME_ALLOW");
+        if (r == -ENOENT)
+                return true;
+
+        return r;
+}
+
+static int naming_sysattr_allowed(sd_device *dev, const char *sysattr) {
+        int r;
+        char *sysattr_property;
+
+        assert(dev);
+        assert(sysattr);
+
+        sysattr_property = strjoina("ID_NET_NAME_ALLOW_", sysattr);
+        ascii_strupper(sysattr_property);
+
+        r = device_get_property_bool(dev, sysattr_property);
+        if (r < 0 && r != -ENOENT)
```

Comment:
This can be simplified to
```c
if (r == -ENOENT)
        /* if  ID_NET_NAME_ALLOW is not set or set to 1 default is to allow */
        return naming_sysattr_allowed_by_default(dev);

return r;
```

Reply:
fixed

Reply:
Still I prefer shorter code suggested by @dtardon:
```c
        r = device_get_property_bool(dev, sysattr_property);
        if (r == -ENOENT)
                /* If ID_NET_NAME_ALLOW is not set or set to 1 default is to allow */
                return naming_sysattr_allowed_by_default(dev);

        return r;
```